pyccl/halos/halo_model.py

```diff
@@ -186,7 +186,7 @@ def number_counts(self, cosmo, *, selection,
             mint[i] = self._integrator(
                 dVda[i] * self._mf[..., :] * _selm[..., :],
                 self._lmass
-            )
+            ).squeeze()
```

Comment:
A few warnings were being triggered because of this (the result of the integral is an array of size 1 instead of a scalar, which numpy doesn't like).